clientjs/rxhtml.js

```diff
@@ -2206,6 +2206,13 @@ var RxHTML = (function () {
         retry_sm.backoff = 1;
         if ("data" in payload.delta) {
           co.tree.update(payload.delta.data);
+          const viewstate = localStorage.getItem('rxviewstate');
+          if (null !== viewstate) {
+            setTimeout(function() {
```

Comment:
@mathgladiator  I am not entirely sure why this setTimeout is needed, but it doesn't work without it (especially for the `rx:expand-view-state` items). I didn't see any new stream data coming in, but I imagine RxHTML has to process the current URL and render the page before the viewstate can be injected. Maybe there is a better way to do this? Will it fail if it takes more than 100ms to load, or does the single nature of JS essentially make this a "next tick" and it will work as is even with large UI's.

Reply:
The place to read the localStorage would be self.run where the init is constructed.